File: service/shoes-service/src/main/java/com/nakick/shoes_service/entity/Sku.java
package com.nakick.shoes_service.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Document(collection = "skus")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Sku {
    
    @Id
    private String id;
    
    @Field("product_id")
    private String productId;
    
    @Field("color")
    private String color;
    
    @Field("size")
    private String size;
    
    @Field("price")
    private BigDecimal price;
    
    @Field("stock")
    private Integer stock;
    
    @Field("image_url")
    private String imageUrl;
    
    @Field("sku_code")
    private String skuCode;
    
    @Field("created_at")
    private LocalDateTime createdAt;
    
    @Field("updated_at")
    private LocalDateTime updatedAt;
    
    @Field("is_active")
    private Boolean isActive;
    
    public Sku(String productId, String color, String size, BigDecimal price, Integer stock, String imageUrl, String skuCode) {
        this.productId = productId;
        this.color = color;
        this.size = size;
        this.price = price;
        this.stock = stock;
        this.imageUrl = imageUrl;
        this.skuCode = skuCode;
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
        this.isActive = true;
    }
    
    public void updateStock(Integer quantity) {
        this.stock = this.stock + quantity;
        this.updatedAt = LocalDateTime.now();
    }
    
    public void decreaseStock(Integer quantity) {
        if (this.stock >= quantity) {
            this.stock = this.stock - quantity;
            this.updatedAt = LocalDateTime.now();
        } else {
            throw new RuntimeException("Insufficient stock. Available: " + this.stock + ", Requested: " + quantity);
        }
    }
    
    public Boolean isInStock() {
        return this.stock > 0 && this.isActive;
    }
    
    public void activate() {
        this.isActive = true;
        this.updatedAt = LocalDateTime.now();
    }
    
    public void deactivate() {
        this.isActive = false;
        this.updatedAt = LocalDateTime.now();
    }
}

File: frontend/mobile/NAKick/lib/models/sku.dart
class Sku {
  final String id;
  final String productId;
  final String color;
  final String size;
  final double price;
  final int stock;
  final String imageUrl;
  final String skuCode;
  final DateTime createdAt;
  final DateTime updatedAt;
  final bool isActive;
  final bool inStock;

  Sku({
    required this.id,
    required this.productId,
    required this.color,
    required this.size,
    required this.price,
    required this.stock,
    required this.imageUrl,
    required this.skuCode,
    required this.createdAt,
    required this.updatedAt,
    required this.isActive,
    required this.inStock,
  });

  static double _toDouble(dynamic value) {
    if (value == null) return 0.0;
    if (value is num) return value.toDouble();
    if (value is String) return double.tryParse(value) ?? 0.0;
    return 0.0;
  }

  factory Sku.fromJson(Map<String, dynamic> json) {
    return Sku(
      id: json['id']?.toString() ?? '',
      productId: json['productId'] ?? json['product_id'] ?? '',
      color: json['color'] ?? '',
      size: json['size']?.toString() ?? '',
      price: _toDouble(json['price']),
      stock: json['stock'] ?? 0,
      imageUrl: json['imageUrl'] ?? json['image_url'] ?? '',
      skuCode: json['skuCode'] ?? json['sku_code'] ?? '',
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : DateTime.now(),
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : DateTime.now(),
      isActive: json['isActive'] ?? json['is_active'] ?? true,
      inStock: json['inStock'] ?? json['in_stock'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'productId': productId,
      'color': color,
      'size': size,
      'price': price,
      'stock': stock,
      'imageUrl': imageUrl,
      'skuCode': skuCode,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
      'isActive': isActive,
      'inStock': inStock,
    };
  }
}
